mbed_devices/_internal/linux/device_detector.py

```diff
@@ -10,4 +14,27 @@ class LinuxDeviceDetector(DeviceDetector):
 
     def find_candidates(self) -> List[CandidateDevice]:
         """Return a list of CandidateDevices."""
-        pass
+        context = pyudev.Context()
+        return [
+            CandidateDevice(
+                mount_points=_find_fs_mounts_for_device(disk.properties.get("DEVNAME")),
+                product_id=disk.properties.get("ID_MODEL_ID"),
+                vendor_id=disk.properties.get("ID_VENDOR_ID"),
+                serial_number=disk.properties.get("ID_SERIAL_SHORT"),
+                serial_port=_find_serial_port_for_device(disk, pyudev.Context()),
+            )
+            for disk in context.list_devices(subsystem="block", ID_BUS="usb")
+        ]
+
+
+def _find_serial_port_for_device(disk: pyudev.Device, context: pyudev.Context) -> Any:
+    """Try to find a serial port associated with the given device."""
+    for tty_dev in context.list_devices(subsystem="tty"):
+        if tty_dev.properties.get("ID_SERIAL_SHORT") == disk.properties.get("ID_SERIAL_SHORT"):
+            return tty_dev.properties.get("DEVNAME")
+    return ""
```

Comment:
Shouldn't it return `None` if it does not find the serial port?

Reply:
Returning the empty string is enough here I think. Is there a reason it should return `None` specifically?

Reply:
30126d2